signer/signer_test.go

```diff
@@ -203,3 +204,37 @@ func TestNoSuitableKeyFound(t *testing.T) {
 		t.Fatalf("expected to fail with no suitable key found but failed with: %v", err)
 	}
 }
+
+func TestGetRandReaderNoHSM(t *testing.T) {
```

Comment:
Won't this test pass regardless of which reader type we get back? You can type assert down to the actual type to prevent this.